File: backend_project/scripts/generate_sitemap.py
"""Generate sitemap.xml with all university URLs from Supabase."""
import requests

SUPABASE_URL = "https://luribqlhnmgslpoqlxmi.supabase.co"
API_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imx1cmlicWxobm1nc2xwb3FseG1pIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODgyNDQzNjcsImV4cCI6MjEwMzgyMDM2N30.IvnGUPVueylQjCkD5UPVtAosM3XKI3KNBkdCf2UbGus"
BASE = "https://findmyuni.pk"

xml = ['<?xml version="1.0" encoding="UTF-8"?>',
       '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">']

# Static pages
for page, freq, pri in [("/", "daily", "1.0"), ("/universities", "daily", "0.9"),
                          ("/compare", "weekly", "0.7"), ("/about", "monthly", "0.5"),
                          ("/contact", "monthly", "0.5"), ("/features", "monthly", "0.5")]:
    xml.append(f'  <url><loc>{BASE}{page}</loc><changefreq>{freq}</changefreq><priority>{pri}</priority></url>')

# University pages
offset = 0
count = 0
while True:
    r = requests.get(f"{SUPABASE_URL}/rest/v1/universities",
                     headers={"apikey": API_KEY, "Authorization": f"Bearer {API_KEY}"},
                     params={"select": "id,updated_at", "offset": offset, "limit": 100})
    if r.status_code not in (200, 206): break
    data = r.json()
    for u in data:
        date = (u.get("updated_at") or "2026-09-03")[:10]
        xml.append(f'  <url><loc>{BASE}/universities/{u["id"]}</loc><lastmod>{date}</lastmod><changefreq>weekly</changefreq><priority>0.8</priority></url>')
        count += 1
    if len(data) < 100: break
    offset += 100

xml.append('</urlset>')

output = "\n".join(xml)
import os
sitemap_path = os.path.join(os.path.dirname(__file__), "..", "..", "scrape-my-uni", "public", "sitemap.xml")
with open(sitemap_path, "w") as f:
    f.write(output)

print(f"Sitemap generated with {count} university URLs + 6 static pages")

File: scrape-my-uni/scripts/generate-sitemap.js
/**
 * Generate sitemap.xml with slug-based URLs
 * Run: node scripts/generate-sitemap.js
 */

import { readFileSync, writeFileSync } from 'fs';
import { join } from 'path';

const SITE_URL = 'https://www.findmyuni.site';

// Blog posts
const blogPosts = [
  'top-10-universities-pakistan-2026',
  'nust-admission-2026-guide-net-test',
  'best-cs-universities-pakistan-2026',
  'university-scholarships-pakistan-2026',
  'public-vs-private-universities-pakistan',
  'write-perfect-university-personal-statement',
  'hec-university-categories-wxyz-explained',
  'engineering-admissions-pakistan-2026-ecat',
  'career-options-pakistani-graduates-2026',
  'pakistani-students-studying-abroad-guide'
];

function generate() {
  const xml = [];
  xml.push('<?xml version="1.0" encoding="UTF-8"?>');
  xml.push('<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">');
  
  // Static pages
  const staticPages = [
    { path: '/', freq: 'daily', pri: '1.0' },
    { path: '/universities', freq: 'daily', pri: '0.9' },
    { path: '/blog', freq: 'weekly', pri: '0.9' },
    { path: '/compare', freq: 'weekly', pri: '0.7' },
    { path: '/about', freq: 'monthly', pri: '0.5' },
    { path: '/contact', freq: 'monthly', pri: '0.5' },
    { path: '/features', freq: 'monthly', pri: '0.5' },
  ];
  
  for (const page of staticPages) {
    xml.push(`  <url><loc>${SITE_URL}${page.path}</loc><changefreq>${page.freq}</changefreq><priority>${page.pri}</priority></url>`);
  }
  
  // Blog posts
  for (const slug of blogPosts) {
    xml.push(`  <url><loc>${SITE_URL}/blog/${slug}</loc><lastmod>2026-09-04</lastmod><changefreq>monthly</changefreq><priority>0.8</priority></url>`);
  }
  
  // University pages (from slug file)
  const slugsPath = join(import.meta.dirname, '..', 'public', 'university-slugs.json');
  try {
    const unis = JSON.parse(readFileSync(slugsPath, 'utf-8'));
    for (const uni of unis) {
      xml.push(`  <url><loc>${SITE_URL}/universities/${uni.slug}</loc><changefreq>weekly</changefreq><priority>0.8</priority></url>`);
    }
    console.log(`Added ${unis.length} university URLs`);
  } catch (e) {
    console.log('Warning: university-slugs.json not found');
  }
  
  xml.push('</urlset>');
  
  const outputPath = join(import.meta.dirname, '..', 'public', 'sitemap.xml');
  writeFileSync(outputPath, xml.join('\n'));
  console.log(`Sitemap generated: ${outputPath}`);
}

generate();
